src/main/java/training/endava/playground/generics/types/LimitedContainer.java

```diff
@@ -0,0 +1,27 @@
+package training.endava.playground.generics.types;
+
+import java.util.ArrayList;
+
+public class LimitedContainer<T, N extends Number> {
+
+    private ArrayList<T> contents;
+
+    private N limit;
+
+    public LimitedContainer(N limit)
+    {
+        contents = new ArrayList<>();
+        this.limit = limit;
+    }
+
+    public void addContainer(T item)
+    {
+        if(this.contents.size() < (Integer) limit){
+            this.contents.add(item);
```

Comment:
no need of this.<field>, it's pretty clear that contents is the current class's field and not a parameter or smth else